File: hollywood/src/test/scala/scalaInversion/NoInversionTest.scala
package scalaInversion
import org.scalatest._
import api.OriginalApi
import api.ApiStatus._

class NoInversionTest extends FunSpec with ShouldMatchers {

  it("fails without initialization") {
    val api = new OriginalApi()
    intercept[IllegalStateException] {
      api.operationOne("test")
    }
  }

  it("works with initialization") {
    val api = new OriginalApi()
    api.expensiveInit()
    val result = api.operationOne("test")
    api.close()
    result.getStatus should be(OK)
    result.getMessage should be("operationOne performed on test")
  }

}

File: fodder/src/test/java/javaInversion/NoInversionTest.java
package javaInversion;

import static javaSupport.api.ApiStatus.OK;
import static org.junit.Assert.assertEquals;
import javaSupport.api.OperationResult;
import javaSupport.api.OriginalApi;

import org.junit.Before;
import org.junit.Test;

public class NoInversionTest {

	private OriginalApi api;

	@Before
	public void setUp() throws Exception {
		api = new OriginalApi();
	}

	@Test(expected = IllegalStateException.class)
	public void itFailsWithoutInitialization() {
		api.operationOne("test");
	}
	
	@Test
	public void itWorksWithInitialization() {
		api.expensiveInit();
		OperationResult result = api.operationOne("test");
		api.close();
		assertEquals(OK, result.getStatus());
		assertEquals("operationOne performed on test", result.getMessage());
	}
}
